src/vm/builtins/builtin_runner/signature.zig

```diff
@@ -56,18 +81,70 @@ pub const SignatureBuiltinRunner = struct {
         };
     }
 
+    pub fn addSignature(self: *Self, relocatable: Relocatable, rs: struct { Felt252, Felt252 }) !void {
+        try self.signatures.put(relocatable, .{
+            .r = rs[0],
+            .s = rs[1],
+        });
+    }
+
     pub fn initSegments(self: *Self, segments: *MemorySegmentManager) !void {
-        _ = self;
-        _ = segments;
+        self.base = @intCast((try segments.addSegment()).segment_index);
     }
 
     pub fn initialStack(self: *Self, allocator: Allocator) !ArrayList(MaybeRelocatable) {
-        _ = self;
         var result = ArrayList(MaybeRelocatable).init(allocator);
         errdefer result.deinit();
+
+        if (self.included) {
+            try result.append(MaybeRelocatable.fromRelocatable(Relocatable.new(@intCast(self.base), 0)));
+        }
+
         return result;
     }
 
+    pub fn base(self: *Self) usize {
```

Comment:
Here we don't need this one as long as `self.base` is public and not private in Zig.